### PR TITLE
Sort job state descending

### PR DESCRIPTION
**Pull request by gottscj**
+ Use objectId instead of string for Job id
   - This will allow sorting by ID
+ Add migration step

### CODE REVIEW

File: src/Hangfire.Mongo/Dto/JobDetailedDto.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace Hangfire.Mongo.Dto
     public class JobDetailedDto
     {
         [BsonId]
```
Comment (MarLoe) [inline src/Hangfire.Mongo/Dto/JobDetailedDto.cs]: Is the BsonId attribute still needed?

File: src/Hangfire.Mongo/Dto/JobDto.cs [hunk 2]
```diff
@@ -8,8 +9,7 @@ namespace Hangfire.Mongo.Dto
 #pragma warning disable 1591
     public class JobDto
     {
-        [BsonId(IdGenerator = typeof(StringObjectIdGenerator))]
-        public string Id { get; set; }
+        public ObjectId Id { get; set; }
```
Comment (MarLoe) [inline src/Hangfire.Mongo/Dto/JobDto.cs]: Here you have removed the BsonId attribute. Let's try to be consequent and do the same everywhere

**Comment by MarLoe** [comment]
Also, we should create a 0.5.6 branch and merge to that.

I'll do that later this weekend :)

File: src/Hangfire.Mongo/Dto/JobDto.cs (hunk 2 above)
Comment (gottscj) [reply src/Hangfire.Mongo/Dto/JobDto.cs]: Ok

**Pull request merged**